lib/manager/sbt/readme.md

```diff
@@ -0,0 +1,245 @@
+## Overview
+
+#### Name of package manager
+
+sbt
+
+---
+
+#### What language does this support?
+
+Scala
+
+---
+
+#### Does that language have other (competing?) package managers?
+
+Maven
+
+## Package File Detection
+
+#### What type of package files and names does it use?
+
+As far as I know, there is 2 package files:
+- One for main dependencies and some configurations (https://www.scala-sbt.org/1.0/docs/Basic-Def.html)
+- One for plugin dependencies and plugin configurations (https://www.scala-sbt.org/1.0/docs/Plugins.html)
+
+---
+
+#### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+
+`['build\\.sbt', 'plugins\\.sbt']`
+
+Note that `plugins.sbt` file is usually in a `project/` subdirectory.
+
+---
+
+#### Is it likely that many users would need to extend this pattern for custom file names?
+
+Not likely.
+
+---
+
+#### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+
+Not likely. The extension `.sbt` should not be used for other files.
+
+## Parsing and Extraction
+
+#### Can package files have "local" links to each other that need to be resolved?
+
+No
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+Scala syntax
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+I think reading the files with Scala would be the easier. Due to the fact that the files are plain Scala code, the
+dependencies can be declared in multiple ways.
+
+But, I guess there is two most commonly used way to declare dependencies and using regex could work for these formats.
+
+Another option might be to use ScalaJS. I don't know it but it has same syntax as Scala and can be read/used within Javascript code.
+
+If we can use Scala or ScalaJS, the idea would be to execute the `build.sbt` or `plugins.sbt` file and work on the `libraryDependencies` or `dependencyOverrides` variables.
+
+_Update (thanks to @ChristianMurphy)_: https://github.com/scalameta/scalameta running on http://www.scala-js.org could be a way to parse Scala without depending on a JDK.
+E.G. https://astexplorer.net/#/gist/027dce11e6927b4ad39ea097ce6289b9/ea4d1048f32063a71d727d01178ae6d01087a62f
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+Dependencies can have a "scope" like `Test` or `Provided`.
+
+Renovate shouldn't care of scopes: they should be kept as is but they don't have any impact on the "new dependency resolution".
```

Comment:
Great, we will just tag the as `depType` so that users have the flexibility to assign rules to them if they need.